### PR TITLE
Allow compute init to be reinvoked within an existing package directory

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Refactors `compute init` allowing it to be reinvoked within an existing package directory. This is allows more frictionless user journeys between the UI and CLI, for instance allowing as user to regenerate domain or backend etc.

Best reviewed with `?w=1` to ignore whitespace changes: https://github.com/fastly/cli/pull/138/files?w=1

### What
- Changes the flag parsing to populate the manifest data-structure for fields `serviceid`, `name`, `description`, `author`
- Guards package template fetching/creation if package already exists
- Allows multiple author flags to be provided
- Adds test for existing manifest

### CODE REVIEW

File: pkg/compute/init.go [hunk 1]
```diff
@@ -52,26 +52,24 @@ var (
...
+	c.manifest.File.Read(manifest.Filename)
 	c.CmdClause = parent.Command("init", "Initialize a new Compute@Edge package locally")
-	c.CmdClause.Flag("service-id", "Existing service ID to use. By default, this command creates a new service").Short('s').StringVar(&c.serviceID)
-	c.CmdClause.Flag("name", "Name of package, defaulting to directory name of the --path destination").Short('n').StringVar(&c.name)
-	c.CmdClause.Flag("description", "Description of the package").Short('d').StringVar(&c.description)
-	c.CmdClause.Flag("author", "Author of the package").Short('a').StringVar(&c.author)
+	c.CmdClause.Flag("service-id", "Existing service ID to use. By default, this command creates a new service").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("name", "Name of package, defaulting to directory name of the --path destination").Short('n').StringVar(&c.manifest.File.Name)
+	c.CmdClause.Flag("description", "Description of the package").Short('d').StringVar(&c.manifest.File.Description)
+	c.CmdClause.Flag("author", "Author of the package").Short('a').StringsVar(&c.manifest.File.Authors)
```
Comment (thommahoney) [inline pkg/compute/init.go:72]:
```suggestion
	c.CmdClause.Flag("author", "Author(s) of the package").Short('a').StringsVar(&c.manifest.File.Authors)
```

File: pkg/app/run_test.go [hunk 2]
```diff
@@ -263,7 +263,7 @@ COMMANDS
     -n, --name=NAME                Name of package, defaulting to directory name
                                    of the --path destination
     -d, --description=DESCRIPTION  Description of the package
-    -a, --author=AUTHOR            Author of the package
+    -a, --author=AUTHOR ...        Author of the package
```
Comment (thommahoney) [inline pkg/app/run_test.go:266]:
```suggestion
    -a, --author=AUTHOR ...        Author(s) of the package
```
(in case you want to commit both suggestions from the GH UI)

File: pkg/compute/init.go [hunk 3]
```diff
@@ -367,24 +376,24 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
...
+	if description != "" {
+		fmt.Fprintf(progress, "Setting description in manifest to %s...\n", description)
+		m.Description = description
 	}
 
-	if c.author != "" {
-		fmt.Fprintf(progress, "Setting author in manifest to %s...\n", c.author)
-		m.Authors = []string{c.author}
+	if len(authors) > 0 {
+		fmt.Fprintf(progress, "Setting author in manifest to %s...\n", strings.Join(authors, ", "))
```
Comment (thommahoney) [inline pkg/compute/init.go:388]:
```suggestion
		fmt.Fprintf(progress, "Setting authors in manifest to %s...\n", strings.Join(authors, ", "))
```

File: pkg/compute/manifest/manifest.go [hunk 4]
```diff
@@ -58,6 +62,32 @@ func (d *Data) ServiceID() (string, Source) {
...
+
+// Authors yields an Authors.
+func (d *Data) Authors() ([]string, Source) {
+	if len(d.Flag.Authors) > 0 {
+		return d.Flag.Authors, SourceFlag
+	}
+
+	if len(d.File.Authors) > 0 {
+		return d.File.Authors, SourceFile
+	}
```
Comment (thommahoney) [inline pkg/compute/manifest/manifest.go:86]: Would it make sense to combine them?

Comment (phamann) [reply pkg/compute/manifest/manifest.go:86]: They are subtly different. One returns the `Flag` value, the other returns the `File` value.
